app/shop.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+import datetime
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: list[int, int],
+            products: dict[str, float],
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    @classmethod
+    def from_dict(
+            cls,
+            shop_dict: dict,
+    ) -> Shop:
+        return cls(
+            shop_dict.get("name"),
+            shop_dict.get("location"),
+            shop_dict.get("products")
+        )
+
+    def service(self, customer_name: str, desired_products: dict) -> float:
+        products_to_buy = {}
+        total_cost = 0
+
+        for product, quantity in desired_products.items():
+            price = quantity * self.products[product]
+            total_cost += price
+
+            if float(price).is_integer():
+                price = int(price)
+            else:
+                round(price, 1)
+            products_to_buy[product] = (quantity, round(price, 1))
+
+        if float(total_cost).is_integer():
+            total_cost = int(total_cost)
+        else:
+            round(total_cost, 1)
```

Comment:
Similarly, the line `round(total_cost, 1)` does not modify `total_cost`. Assign the result back to `total_cost` to ensure it is rounded to one decimal place.